### PR TITLE
Apply more structure to pod data dirs

### PR DESCRIPTION
**Pull request by thockin**
This is makes it possible to read back "known" pods from disk without
getting other (non-pod) kubelet dirs in the mix.  Ditto for containers
within a pod.  This is just saner overall.  Pods now nest in a pods/
dir.  Likewise containers.  Backwards compat code for now, but should
be stripped out before 1.0.

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -170,16 +170,30 @@ func (kl *Kubelet) GetRootDir() string {
...
 // specified pod.  This directory may not exist if the pod does not exist.
 func (kl *Kubelet) GetPodDir(podUID string) string {
-	return path.Join(kl.GetRootDir(), podUID)
+	// Backwards compat.  The "old" stuff should be removed before 1.0
+	// release.  The thinking here is this:
+	//     !old && !new = use new
+	//     !old && new  = use new
+	//     old && !new  = use old
+	//     old && new   = use new (but warn)
+	oldPath := path.Join(kl.GetRootDir(), podUID)
```
Comment (jbeda) [inline pkg/kubelet/kubelet.go]: I'm confused -- these look like they are the same path.

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -170,16 +170,30 @@ func (kl *Kubelet) GetRootDir() string {
...
-// TODO(thockin): For now, this is the same as the root because that is assumed
-// in other code.  Will fix.
 func (kl *Kubelet) GetPodsDir() string {
-	return kl.GetRootDir()
+	return path.Join(kl.GetRootDir(), "pods")
 }
 
 // GetPodDir returns the full path to the per-pod data directory for the
 // specified pod.  This directory may not exist if the pod does not exist.
 func (kl *Kubelet) GetPodDir(podUID string) string {
```
Comment (jbeda) [inline pkg/kubelet/kubelet.go]: I worry about security here.  A sanity check that there are no special characters here would help.  This is the type of code that can often lead to a directory traversal attack.

**Comment by jbeda** [comment]
LGTM -- the extra security check would be good but the code already is missing it so this is a no-op from that point of view.

Feel free to add something or self merge.

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (thockin) [inline pkg/kubelet/kubelet.go]: I'm not sure I follow.  GetRootDir() is controlled by kubelet, so any trickery would have to come through the flag that controls root dir.  If you are setting flags you are already root.  The joining of pod UID and container name predicate on the validation done before we accept a pod.

Can you clarify what sort of attack you are conceiving of?  Caveat, I suck at security analysis.

Comment (jbeda) [inline pkg/kubelet/kubelet.go]: I'm mostly worried about the container name (and perhaps the UID).  While I know we validate that higher up, defense in depth is the name of the game.

**Pull request merged**